xgcm/grid.py

```diff
@@ -1145,6 +1145,33 @@ def derivative(self, da, axis, **kwargs):
         dx = self.get_metric(diff, ("X",))
         return diff / dx
 
+    @docstrings.dedent
+    def integrate(self, da, axis, **kwargs):
+        """
+        Integrate along specified axis or axes.
+        Grid metrics (e.g. cell length, area, volume) are determined automatically.
+
+        Parameters
+        ----------
+        axis : str, list of str
+            Name of the axis on which to act
+        %(neighbor_binary_func.parameters.no_f)s
+
+        Returns
+        -------
+        da_i : xarray.DataArray
+            The integrated data
+        """
+
+        weight = self.get_metric(da, _maybe_promote_str_to_list(axis))
```

Comment:
you don't need to call _maybe_promote_str_to_list here. It is done inside get_metric.